akka-stream/src/main/scala/akka/stream/SystemMaterializer.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import akka.actor.ActorSystem
+import akka.actor.Deploy
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.annotation.InternalApi
+import akka.stream.impl.MaterializerGuardian
+
+import scala.concurrent.Await
+import scala.concurrent.Promise
+import akka.util.JavaDurationConverters._
+import akka.util.Timeout
+
+import com.github.ghik.silencer.silent
+
+/**
+ * The system materializer is a default materializer to use for most cases running streams, it is a single instance
+ * per actor system that is tied to the lifecycle of that system.
+ *
+ * Not intended to be manually used in user code.
+ */
+object SystemMaterializer extends ExtensionId[SystemMaterializer] with ExtensionIdProvider {
+  override def get(system: ActorSystem): SystemMaterializer = super.get(system)
+
+  override def lookup = SystemMaterializer
+
+  override def createExtension(system: ExtendedActorSystem): SystemMaterializer =
+    new SystemMaterializer(system)
+}
+
+final class SystemMaterializer(system: ExtendedActorSystem) extends Extension {
+  private val systemMaterializerPromise = Promise[Materializer]()
+
+  // load these here so we can share the same instance across materializer guardian and other uses
+  /**
+   * INTERNAL API
+   */
+  @InternalApi @silent("deprecated")
+  private[akka] val materializerSettings = ActorMaterializerSettings(system)
+
+  private implicit val materializerTimeout: Timeout =
+    system.settings.config.getDuration("akka.stream.materializer.creation-timeout").asScala
+
+  //@InternalApi @silent("deprecated")
+  system.systemActorOf(
+    MaterializerGuardian
+      .props(systemMaterializerPromise, materializerSettings)
+      .withDeploy(Deploy.local),
+    "Materializers")
```

Comment:
How about not backporting the guardian but just creating an ActorMaterializer here? Then we can avoid the blocking and having to backport the extension fixes and not breaking Akka.js. etc?

Reply:
Makes sense to me :+1: - wanted to stay somewhat faithful to the 2.6 implementation but it's probably causing more problems than it solves here